packages/react-server/core/components/RootElement.js

```diff
@@ -39,6 +48,16 @@ class RootElement extends React.Component {
 		if (this.props.unsubscribe) this.props.unsubscribe();
 	}
 
+	componentWillReceiveProps(nextProps) {
+		// Incase we receive new props such as during client transitions
+		// We will want to update our state's childProp with any new childProps
+		// that may have been passed in. This still respects props as the ultimate source of truth
+		const newChildProps = _.assign({}, this.state.childProps, nextProps.childProps);
```

Comment:
One downside to doing this is that if `nextProps.childProps` is empty, then `newChildProps` will still retain old keys/values from `this.state.childProps`. The workaround to this would be to pass in `null` as the new value for those keys if that behavior is desired.